bdpy/dl/torch/models.py

```diff
@@ -367,3 +367,447 @@ def forward(self, z: torch.Tensor) -> torch.Tensor:
         g = self.deconv(f)
 
         return g
+
+
+class AlexNetPool5Generator(nn.Module):
+    """ From caffe pool5 generator of bvlc_reference_caffenet.
+    The model trained by DeepSim using ILSVRC dataset.
+    Provided by Alexey Dosovitskiy.
+    """
+
+    def __init__(
+            self, device: Optional[Union[str, Sequence[str]]] = None):
+        super(AlexNetPool5Generator, self).__init__()
+
+        if device is None:
+            self.__device0 = 'cpu'
+            self.__device1 = 'cpu'
+        elif isinstance(device, str):
+            self.__device0 = device
+            self.__device1 = device
+        else:
+            self.__device0 = device[0]
+            self.__device1 = device[1]
+        # input 256x6x6
+        self.Rconv6 = nn.Conv2d(256, 512, kernel_size=3, stride=1, padding=1)
+        self.Rrelu6 = nn.LeakyReLU(0.3, inplace=True)
+        self.Rconv7 = nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=1)
+        self.Rrelu7 = nn.LeakyReLU(0.3, inplace=True)
+        self.Rconv8 = nn.Conv2d(512, 512, kernel_size=3, stride=1, padding=0)
+        self.Rrelu8 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv5 = nn.ConvTranspose2d(
+            512, 256, kernel_size=4, stride=2, padding=1, bias=True)
+        self.relu_deconv5 = nn.LeakyReLU(0.3, inplace=True)
+        self.conv5_1 = nn.ConvTranspose2d(
+            256, 512, kernel_size=3, stride=1, padding=1, bias=True)
+        self.relu_conv5_1 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv4 = nn.ConvTranspose2d(
+            512, 256, kernel_size=4, stride=2, padding=1, bias=True)
+        self.relu_deconv4 = nn.LeakyReLU(0.3, inplace=True)
+        self.conv4_1 = nn.ConvTranspose2d(
+            256, 256, kernel_size=3, stride=1, padding=1, bias=True)
+        self.relu_conv4_1 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv3 = nn.ConvTranspose2d(
+            256, 128, kernel_size=4, stride=2, padding=1, bias=True)
+        self.relu_deconv3 = nn.LeakyReLU(0.3, inplace=True)
+        self.conv3_1 = nn.ConvTranspose2d(
+            128, 128, kernel_size=3, stride=1, padding=1, bias=True)
+        self.relu_conv3_1 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv2 = nn.ConvTranspose2d(
+            128, 64, kernel_size=4, stride=2, padding=1, bias=True)
+        self.relu_deconv2 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv1 = nn.ConvTranspose2d(
+            64, 32, kernel_size=4, stride=2, padding=1, bias=True)
+        self.relu_deconv1 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv0 = nn.ConvTranspose2d(
+            32, 3, kernel_size=4, stride=2, padding=1, bias=True)
+        self.rconv = nn.Sequential(
```

Comment:
It would be better if the parent class and the instance variable `nn.Sequential` did not have references to the same module. For example, currently, `gene.Rconv6` and `gene.rconv.get_submodule('0')` have the exact same reference.